specs-jetpack-calypso/wp-jetpack-onboarding-spec.js

```diff
@@ -0,0 +1,432 @@
+				this.url = url;
+			} );
+		} );
+
+		test.it( 'Can navigate to onboarding flow', function() {
+			return driver.get( this.url + onboardingUrlExt );
+		} );
+
+		test.it( 'Can fill out site title and tagline', function() {
+			const siteTitleTaglinePage = new SiteTitleTaglinePage( driver );
+			siteTitleTaglinePage.enterTitle( blogTitle );
+			siteTitleTaglinePage.enterTagline( blogTagline );
+			return siteTitleTaglinePage.selectContinue();
+		} );
+
+		test.it( 'Can select Business Site', function() {
+			const siteTypePage = new SiteTypePage( driver );
+			return siteTypePage.selectBusinessSite();
+		} );
+
+		test.it( 'Can select static homepage', function() {
+			const setHomepagePage = new SetHomepagePage( driver );
+			return setHomepagePage.selectPage();
+		} );
+
+		test.it( 'Can skip add a contact form', function() {
+			const wizardNavigationComponent = new WizardNavigationComponent( driver );
+			return wizardNavigationComponent.skipStep();
+		} );
+
+		test.it( 'Can skip add a business address', function() {
+			const wizardNavigationComponent = new WizardNavigationComponent( driver );
+			return wizardNavigationComponent.skipStep();
+		} );
+
+		test.it( 'Can make business an online store', function() {
+			const installWooCommercePage = new InstallWooCommercePage( driver );
+			return installWooCommercePage.selectSellOnline();
+		} );
+
+		test.it( 'Can select activate on activate stats page', function() {
+			const activateStatsPage = new ActivateStatsPage( driver );
+			return activateStatsPage.selectActivateStats();
+		} );
+
+		test.it( 'Can approve connection on the authorization page', function() {
+			const jetpackAuthorizePage = new JetpackAuthorizePage( driver );
+			return jetpackAuthorizePage.approveConnection();
+		} );
+
+		test.it( 'Can select activate on activate stats page', function() {
+			const activateStatsPage = new ActivateStatsPage( driver );
+			return activateStatsPage.selectContinue();
+		} );
+
+		test.it( 'Can see onboarding summary page', function() {
+			const summaryPage = new SummaryPage( driver );
+			return summaryPage.countToDoSteps()
+				.then( toDoCount => assert.equal( toDoCount, 2, 'Expected and actual steps are not equal.' ) )
+				.then( () => summaryPage.selectVisitSite() );
+		} );
+
+		test.it( 'Can see site home page', function() {
+			const viewSitePage = new ViewSitePage( driver );
+			return viewSitePage.siteTitle()
+				.then( title => assert.equal( title.toUpperCase(), blogTitle.toUpperCase(), 'Site title not is not correct' ) )
+				.then( () => viewSitePage.siteTagline() )
+				.then( tagline => assert.equal( tagline, blogTagline, 'Site tagline not is not correct' ) );
+		} );
+	} );
+
+	test.describe( 'Onboard personal site that has already been connected: @parallel @jetpack', function() {
+		this.bailSuite( true );
+		const blogTitle = dataHelper.randomPhrase();
+		const blogTagline = dataHelper.randomPhrase();
+
+		test.it( 'Can create wporg site', function() {
```

Comment:
This connection flow could be replaced with `JetpackConnectFlow#connectFromWPAdmin`